src/Microsoft.AspNetCore.Razor.Language/RazorProjectEngineBuilderExtensions.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+
+namespace Microsoft.AspNetCore.Razor.Language
+{
+    public static class RazorProjectEngineBuilderExtensions
+    {
+        public static void ReplaceFeature<TFeature>(this RazorProjectEngineBuilder builder, IRazorFeature feature) where TFeature : IRazorFeature
```

Comment:
Just discussed with @NTaylorMullen. The user shouldn't need to call this method to add a feature for the first time.